### PR TITLE
Fluid inspector and cleanups

### PR DESCRIPTION
**Pull request by huwshimi**
Made the inspector adapt to the height of the content. Cleaned up duplicate, old and redundant styles.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- poke around the inspector
- check that the height of the inspector adapts to the height of the content (this will be improved during the react rebuild)
- check that nothing looks broken following the styles cleanup

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2821/
Test PASSed.

File: jujugui/static/gui/src/app/assets/css/_content-panel.scss [hunk 1]
```diff
@@ -1,4 +1,7 @@
-.content-panel {
+// This is the content panel that slides out from the left that contains
+// charm details etc.
+.bws-view-data {
+    @extend .customize-scrollbar;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/_content-panel.scss]: Could you use a placeholder like %customize-scrollbar

File: jujugui/static/gui/src/app/assets/css/_content-panel.scss [hunk 2]
```diff
@@ -10,6 +13,57 @@
     background-image: none;
     box-shadow: 0 0 2px rgba(0, 0, 0, 0.3);
 
+    .left-breakout {
+        @extend .type9;
+        z-index: 500;
+        left: auto;
+        right: $inspector-width;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/_content-panel.scss]: Do you need z-index, left and right here as they dont seem to change anything once removed?

File: jujugui/static/gui/src/app/assets/css/_content-panel.scss [hunk 3]
```diff
@@ -10,6 +13,57 @@
...
+
+        .tab-panels {
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/_content-panel.scss]: Are there tab-panels in the unit view?

File: jujugui/static/gui/src/app/assets/css/_content-panel.scss [hunk 4]
```diff
@@ -10,6 +13,57 @@
...
+            top: 97px;
+        }
+
+        .charm .yui3-juju-browser-tabview {
+            margin-top: 30px;
+        }
+
+        .unit-overview {
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/_content-panel.scss]: I find this used on the unit details panel. How do I get this to view?

File: jujugui/static/gui/src/app/assets/css/_mixins.scss [hunk 5]
```diff
@@ -142,14 +142,11 @@
   display: block;
   background-color: $color;
 }
-.bws-list {
+
+%bws-list {
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/_mixins.scss]: Nice one

File: jujugui/static/gui/src/app/assets/css/browser/_added-services-button.scss [hunk 6]
```diff
@@ -1,6 +1,4 @@
 .added-services-button {
-  position: absolute;
-  top: 81px;
   line-height: 58px;
   padding-left: 20px;
   width: 270px;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/browser/_added-services-button.scss]: Don't think this needs a width as its a block object.

File: jujugui/static/gui/src/app/assets/css/inspector/_juju-inspector.scss [hunk 7]
```diff
@@ -215,6 +74,22 @@
             &.wide {
                 width: 100%;
             }
+
+            &.confirm {
+                float: right;
+
+                &,
+                &:hover {
+                    background-color: $confirm-button-color;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/inspector/_juju-inspector.scss]: The confirm button should have a hover background of #ae3910 with a transition of background .3s

**Comment by anthonydillon** [comment]
QA ok with a couple of questions in the code.

File: jujugui/static/gui/src/app/assets/css/_content-panel.scss (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/css/_content-panel.scss]: Done.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2828/
Test FAILed.

**Comment by huwshimi** [comment]
The last build was successful: http://ci.jujugui.org:8080/job/juju-gui/2829/

**Comment by huwshimi** [comment]
QA only. As discussed in person there are three followups:
- relation lines don't quite touch the edges of the circle
- the unit statuses are not reflected by the circle border
- the relation line snapping needs to be removed

**Comment by huwshimi** [comment]
Thanks :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**